### PR TITLE
Prepare for emscripten-releases uploads

### PR DESCRIPTION
**Pull request by kripken**
Allow detecting if we are the emscripten-releases bot, and upload to the proper bucket based on that.

### CODE REVIEW

File: src/buildbot.py [hunk 1]
```diff
@@ -48,6 +57,12 @@ def BuildNumber():
   return BUILDBOT_BUILDNUMBER
 
 
+def IsEmscriptenReleasesBot():
+  """Return true if running on the emscripten-releases builders,
+     False otherwise."""
+  return BUILDBOT_MASTERNAME is EMSCRIPTEN_RELEASES_BOT
```
Comment (dschuff) [inline src/buildbot.py]: let's use `==` here for consistency with the rest of this repo.

Comment (kripken) [reply src/buildbot.py]: Done.

File: src/buildbot.py [hunk 2]
```diff
@@ -37,6 +37,15 @@
...
 BUILDBOT_BUILDNUMBER = os.environ.get('BUILDBOT_BUILDNUMBER', None)
 BUILDBOT_BUILDERNAME = os.environ.get('BUILDBOT_BUILDERNAME', None)
+BUILDBOT_MASTERNAME = os.environ.get('BUILDBOT_MASTERNAME', None)
+
+# Possible masters include None (running locally), the waterfall integration
+# bot, or the emscripten-releases bot.
+WATERFALL_BOT = 'client.wasm.llvm'
+EMSCRIPTEN_RELEASES_BOT = 'emscripten-releases'
+
+assert BUILDBOT_MASTERNAME in [None, WATERFALL_BOT, EMSCRIPTEN_RELEASES_BOT], \
```
Comment (dschuff) [inline src/buildbot.py]: I realized we'll also need to pass the info about whether the build is a CI build or a try build, and only try to upload from CI builds. I added `BUILDBOT_BUCKET` to the env (which IIUC should be either "ci" or "try").

Comment (kripken) [reply src/buildbot.py]: Thanks, I'll open a PR.

**Pull request merged**